cmd/controller/main.go

```diff
@@ -196,8 +199,8 @@ func main() {
 	fleetController := fleets.NewController(wh, health, kubeClient, extClient, agonesClient, agonesInformerFactory)
 	faController := fleetallocation.NewController(wh, allocationMutex,
 		kubeClient, extClient, agonesClient, agonesInformerFactory)
-	gasController := gameserverallocations.NewController(wh, health, gsCounter, topNGSForAllocation,
-		kubeClient, extClient, agonesClient, agonesInformerFactory)
+	gasController := gameserverallocations.NewController(api, health, gsCounter, topNGSForAllocation,
```

Comment:
This is OK for now but at some point we should run the allocation controller and api server in it's own Pod(s).

Reply:
Yeah totally agreed - but we can do that as a separate PR :+1:

Reply:
+1 to its own set of pods with higher number of replicas, so that this service is never disrupted when we upgrade Agones itself. This is going to be critical for production use.